test/restart.jl

```diff
@@ -0,0 +1,144 @@
+                                Dict(
+                                    "restart_file" => restart₁,
+                                    "output_dir" => joinpath(
+                                        output_loc,
+                                        "restart_test_output",
+                                    ),
+                                ),
+                            ),
+                        )
+                        simulation_test₁ = CA.get_simulation(config₁)
+                        @info "Advancing restarted simulation"
+                        CA.solve_atmos!(simulation_test₁)
+                        @info "Restarted simulation complete"
+
+                        # TODO: Extend for MPI and GPU checks as part of standard runtests.jl
+                        # TODO: Fix output paths once `OutputPathGenerator` is restored.
+                        file0 = joinpath(
+                            output_loc,
+                            "gen_test_output/output_active/day0.0.hdf5",
+                        )
+                        f0 = CC.InputOutput.HDF5Reader(
+                            file0,
+                            ClimaComms.SingletonCommsContext(),
+                        )
+                        f0Y = CC.InputOutput.read_field(f0, "Y")
+
+                        file1 = joinpath(
+                            output_loc,
+                            "gen_test_output/output_active/day0.3.hdf5",
+                        )
+                        file2 = joinpath(
+                            output_loc,
+                            "restart_test_output/output_active/day0.3.hdf5",
+                        )
+                        f1 = CC.InputOutput.HDF5Reader(
+                            file1,
+                            ClimaComms.SingletonCommsContext(),
+                        )
+                        f2 = CC.InputOutput.HDF5Reader(
+                            file2,
+                            ClimaComms.SingletonCommsContext(),
+                        )
+                        f1Y = CC.InputOutput.read_field(f1, "Y")
+                        f2Y = CC.InputOutput.read_field(f2, "Y")
+                        close(f1)
+                        close(f2)
+                        FT = eltype(f1Y)
+                        @info "Check file-read from checkpoint data"
+                        @test maximum(
+                            abs.(parent(f1Y.c.ρ) .- parent(f2Y.c.ρ)),
+                        ) == FT(0)
+                        @test maximum(
+                            abs.(parent(f1Y.c.ρe_tot) .- parent(f2Y.c.ρe_tot)),
+                        ) == FT(0)
+                        @test maximum(
+                            abs.(parent(f1Y.f.u₃) .- parent(f2Y.f.u₃)),
+                        ) <= sqrt(eps(FT))
+                        @test maximum(
+                            abs.(parent(f1Y.c.uₕ) .- parent(f2Y.c.uₕ)),
+                        ) <= sqrt(eps(FT))
```

Comment:
This might be condensed to
```suggestion
                        @test maximum(
                            abs.(parent(f1Y.c) .- parent(f2Y.c)),
                        ) <= sqrt(eps(FT))
                        @test maximum(
                            abs.(parent(f1Y.f) .- parent(f2Y.f)),
                        ) <= sqrt(eps(FT))                        
```

Reply:
Indeed, this was the prior setup - I was curious to check scalar vs vector differences (this needs to be looked at within the ClimaCore HDF5Reader / Writer setups)

Reply:
Ah, got it!